### PR TITLE
back Panel added

### PR DESCRIPTION
**Pull request by annamharikrishna**

### CODE REVIEW

File: cabinet-studio/js/core/cabinet/panel/BackPanel.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    };
+
+    OopsUtils.inherit(BackPanel, SidePanel);
+
+    BackPanel.prototype.getMaximumL = function () {
+        return this.parent.getH();
+    };
+
+    BackPanel.prototype.getL = function () {
+        return this.parent.getH() + (2 * this.grooveOffset);
```
Comment (mithuntnt) [inline cabinet-studio/js/core/cabinet/panel/BackPanel.js]: Grove is different for side panels and partition. Sides have 7 + 1 mm grove, partition have 5 + 1 mm groove.

File: cabinet-studio/js/core/cabinet/panel/BackPanelMaterial.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+define([
+    "./../../utils/oops",
+    "./Material",
+    "./Texture",
+    "./RawMaterial",
+    "./../../Lamination"
+], function (OopsUtils, Material, Texture, RawMaterial, Lamination) {
+    "use strict";
+    var BackPanelMaterial = function BackPanelMaterial(thickness) {
+        var matTexture = new Texture("/godavari/cabinet-studio/images/raw/mdf.jpg", 500, 500);
```
Comment (mithuntnt) [inline cabinet-studio/js/core/cabinet/panel/BackPanelMaterial.js]: BackPanel material is kind of odd. All material should be taken from existing materials.

File: cabinet-studio/js/testcases/core/BackPanel.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
+/**
+ * Created by Techyard Systems PVT LTD on 9/25/2015.
+ */
```
Comment (mithuntnt) [inline cabinet-studio/js/testcases/core/BackPanel.js]: Incorrect file header.

File: cabinet-studio/js/core/Entity.js [hunk 4]
```diff
@@ -25,6 +26,8 @@ define([
         this.cache = {};
         this.joints = {};
         this.isCacheStale = true;
+        this.eventEmitter = new EventEmitter();
```
Comment (kishore25kumar) [inline cabinet-studio/js/core/Entity.js]: Use libraries globally. Don't include them in custom files

**Pull request closed without merging**